feat(treesitter): #trim! can trim all whitespace (#30085)
neovim/neovim: runtime/doc/treesitter.txt

            (#gsub! @_node ".*%.(.*)" "%1")
<
    `trim!`                                          *treesitter-directive-trim!*
        Trims whitespace from the node. Sets a new
        `metadata[capture_id].range`. Trims blank lines (lines that contain
        only whitespace, or are empty) from the end of the node by default
        (for backwards compatibility). Trims all whitespace from both sides of
        the node if parameters are given.

        Parameters: ~
            {capture_id}
            {trim_start_linewise}
            {trim_start_charwise}

>             {trim_end_linewise} (default `true` if only given {capture_id})

Author: I think docs are in a good spot now, lmk if you disagree

            (#gsub! @_node ".*%.(.*)" "%1")
<
    `trim!`                                          *treesitter-directive-trim!*
        Trim whitespace from the node. This will set a new
        `metadata[capture_id].range`. Trims blank lines from the end of the node
        by default (for backwards compatibility). Trims all whitespace from both

>         sides of the node if the optional parameter "1" is given.

Reviewer: I was actually envisioning this to work like the `#offset!` directive, where you can control each of the four(!) sides individually.

Author: Sure gimme a sec- I assume startline startcols endline endcols?

            (#gsub! @_node ".*%.(.*)" "%1")
<
    `trim!`                                          *treesitter-directive-trim!*
        Trim whitespace from the node. This will set a new

>         `metadata[capture_id].range`. The directive takes an optional
>         parameter following the capture group to determine whether to trim all
>         whitespace from both sides of the node (when set to `1`), or to only
>         trim blank lines from the end of the node (this is the default, for
>         backwards compatibility).

Reviewer 2: ```suggestion
        `metadata[capture_id].range`. Trims blank lines from the end of the node
        by default (for backwards compatibility). Trims all whitespace from both
        sides of the node if the optional parameter "1" is given.
```

            (#gsub! @_node ".*%.(.*)" "%1")
<
    `trim!`                                          *treesitter-directive-trim!*

>         Trim whitespace from the node. This will set a new

Reviewer 2: ```suggestion
        Trim whitespace from the node. Sets a new
```

        `metadata[capture_id].range`. Trims blank lines from the end of the node
        by default (for backwards compatibility). Trims all whitespace from both

>         sides of the node if the optional parameter "1" is given.

Reviewer 2: this doc needs to be updated now, right?
```suggestion
        sides of the node if parameters are given.
```

Author: It does; sorry maybe I should mark as draft (even though this is 90% done?)- still working through a test failure which is causing me headache

            (#gsub! @_node ".*%.(.*)" "%1")
<
    `trim!`                                          *treesitter-directive-trim!*
        Trims whitespace from the node. Sets a new

>         `metadata[capture_id].range`. Trims blank lines from the end of the

Reviewer 2: is it blank lines or only empty lines?
```suggestion
        `metadata[capture_id].range`. Trims empty lines from the end of the
```

Reviewer: Whichever can contain (only) whitespace ;)

Author: Yes, so blank- I'll clarify here
